pysb/importers/bngl.py

```diff
@@ -358,24 +378,34 @@ def _parse_rules(self):
     def _parse_expressions(self):
         expr_namespace = dict(
             self.model.parameters | self.model.expressions_constant()
-            | self.model.observables
         )
 
         for e in self._x.iterfind(_ns('{0}ListOfFunctions/{0}Function')):
-            if e.find(_ns('{0}ListOfArguments/{0}Argument')) is not None:
-                self._warn_or_except('Function %s is local, which is not '
-                                     'supported in PySB' % e.get('id'))
+            is_local = False
+            for arg in e.iterfind(_ns('{0}ListOfArguments/{0}Argument')):
+                is_local = True
+                tag_name = arg.get('id')
+                try:
+                    self.model.tags[tag_name]
+                except KeyError:
+                    tag = self.tag(tag_name)
+                    expr_namespace[tag_name] = tag
             expr_name = e.get('id')
             expr_text = e.find(_ns('{0}Expression')).text.replace('^', '**')
             expr_val = 0
             try:
                 expr_val = parse_expr(expr_text, local_dict=expr_namespace)
             except Exception as ex:
+                try:
+                    msg = ex.message
```

Comment:
`ex.message` doesn't exist in Python 3, the replacement is `str(ex)` which is backwards-compatible with Python 2. I guess the try/except would then need to be replaced with `if not msg:`.

Reply:
Replaced with `str(msg)` as suggested